src/components/container/MostRecentNotification/MostRecentNotification.tsx

```diff
@@ -1,27 +1,49 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState } from 'react'
 import "./MostRecentNotification.css"
-import MyDataHelps, { NotificationQueryParameters, NotificationType, Notification } from "@careevolution/mydatahelps-js"
-import { Action, SingleNotification } from '../../presentational'
+import MyDataHelps, { NotificationQueryParameters, NotificationType, Notification, QueryParameters } from "@careevolution/mydatahelps-js"
+import { Action, LoadingIndicator, SingleNotification } from '../../presentational'
 import { add } from 'date-fns'
-import { previewNotification } from './MostRecentNotification.previewdata'
+import { previewNotifications } from './MostRecentNotification.previewdata'
 import language from '../../../helpers/language'
+import { useInitializeView } from '../../../helpers'
 
 export interface MostRecentNotificationProps {
 	notificationType?: NotificationType,
+	notificationIdentifierRegex?: RegExp;
 	onViewMore?: Function,
 	hideAfterHours?: number,
 	previewState?: MostRecentNotificationPreviewState
 	innerRef?: React.Ref<HTMLDivElement>
 }
 
-export type MostRecentNotificationPreviewState = "Default";
+export type MostRecentNotificationPreviewState = 'loading' | 'loaded with data' | 'loaded with no data';
 
 export default function (props: MostRecentNotificationProps) {
 	const [notification, setNotification] = useState<Notification | null>(null);
+	const [loading, setLoading] = useState<boolean>(true);
 
-	function initialize() {
-		if (props.previewState == "Default") {
-			setNotification(previewNotification);
+	async function initialize() {
+		setLoading(true);
+		if (props.previewState === 'loading') {
+			return;
+		}
+
+		if (props.previewState === 'loaded with data') {
+			const typeFilteredNotifications = previewNotifications.filter(n => !props.notificationType || n.type === props.notificationType);
+			const typeAndIdentifierFilteredNotifications = typeFilteredNotifications.filter(n => !props.notificationIdentifierRegex || n.identifier.match(props.notificationIdentifierRegex));
+			const hideAfterHoursFilteredNotifications = typeAndIdentifierFilteredNotifications.filter(n => !props.hideAfterHours || new Date(n.sentDate) > add(new Date(), { hours: -1 * props.hideAfterHours }));
+			if (hideAfterHoursFilteredNotifications.length > 0) {
+				const previewNotification = hideAfterHoursFilteredNotifications[0];
+				setNotification(previewNotification);
+			} else {
+				setNotification(null);
+			}
+			setLoading(false);
+			return;
+		}
+
+		if (props.previewState === 'loaded with no data') {
+			setLoading(false);
```

Comment:
I think you need to do... at least a `setNotification(null)` here or something?  This doesn't appear to be working in the story, I just get the same view as `'loaded with data'`.

Reply:
yep! Fixed